### PR TITLE
RHEL 7.9 Build Fix

### PR DESCRIPTION
**Pull request by naimulhq**
1. In src/main.c , #include <sys/ioctl.h> was missing. This would cause an error for "struct winsize w" stating that "storage size of w isn't known"

2. Even after installing the necessary packages, there were certain .pc files missing that would cause cmake to fail. These files are uploaded in misc_files

3. After installing all the necessary -devel packages, there were certain header files missing that needed to be added to /usr/include or /usr/include/PCSC or /usr/include/editline. These files are in misc_files

4. The help2man package is not available in the RHEL Repositories including EPEL. You can install cmake but it does not give the version required by cmake. 

5. Additional CMAKE_C_FLAGS were added

6. ld fails to find libedit and libpcsclite so symbolic links needed to be created that point to the .so files. The name of the link was determined by running ld -l <package-name> --verbose

I provided a build_hsm_script that is responsible for doing all this in the environment that I am currently working on. No guarantee that it will work for all RHEL environments but it could still be helpful

### CODE REVIEW

**Comment by qpernil** [comment]
Many thanks for investigating this. However, I don't think it's appropriate to have external header files in a PR. What would make more sense is instructions (and links) where to find those files in a 'BUILD on RHEL 7.9' readme file.I'd be glad to include such a file in the repo (with attribution of course).

Also, you still have a small change to main.c, besides the include. Probably not what you intended.
Again, thx for your contribution!

**Comment by qpernil** [comment]
Another question - what version of cmake is available in RHEL 7.9 ?

**Comment by qpernil** [comment]
And another - could you show the errors you encountered without disabling the 3 warning types that you added ? I think it might be better to fix the code than turning off to many warnings. Turning off warnings might bite us later. Also, I find it odd that such errors would not be caught on other platforms. Maybe they are a consequence of you (maybe?) using different header files ?

**Comment by naimulhq** [comment]
I am sorry about that. I could make a README files explaining the header files that are needed. The main reason I included the header files is because I can easily run a script that builds the tools and replaces the missing header files instead of having to do it manually. In addition, the only intended change was the #include. The rest were not intended. The version of cmake available on RHEL 7.9 is 2.8.12.2. When I get the chance, I can rebuild the tools and keep the warnings on. I will paste it here.

**Comment by qpernil** [comment]
Just a thought, couldn't your script get the files from urls instead, using wget or curl for example ?
I will look into the cmake version requirements, but it's unlikely we can get by on such an old version since we are using features that are newer in our build script. Going back this far would make the cmakefiles uglier.
Thx for your feedback.

**Comment by naimulhq** [comment]
It would be possible in any other instances but our environment does not allow any internet access which is why wget or curl is not used.

**Comment by naimulhq** [comment]
I have not forgotten about this. I have just gotten really busy but I will try to get this to you ASAP.

File: CMakeLists.txt [hunk 1]
```diff
@@ -142,6 +142,9 @@ else()
   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-format-zero-length")
   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-missing-braces")
   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-missing-field-initializers")
+  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-variable")
+  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-switch")
+  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-implicit-function-declaration")
```
Comment (chuckatkins) [inline CMakeLists.txt:147]: What's the use case for adding `-Wno-implicit-function-declaration`?  Wouldn't it be particularly useful and reveal missing includes that sholld be prsent?

**Comment by qpernil** [comment]
Hi, 
I have created https://github.com/Yubico/yubihsm-shell/pull/283 that I think includes the changes that are appropriate to do in this repo. Could you let me know if they fix all issues not related to missing packages and headers ?

**Pull request closed without merging**